Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/batch/types/Datastream.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/batch/types/Datastream.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/batch/types/DigitalObject.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/batch/types/DigitalObject.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ArrayInputPanel.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ArrayInputPanel.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/BooleanInputPanel.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/BooleanInputPanel.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ByteArrayInputPanel.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ByteArrayInputPanel.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/Console.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/Console.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ConsoleCommand.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ConsoleCommand.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ConsoleCommandInvoker.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ConsoleCommandInvoker.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ConsoleSendButtonListener.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ConsoleSendButtonListener.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/DateTimeInputPanel.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/DateTimeInputPanel.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/InputPanel.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/InputPanel.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/InputPanelFactory.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/InputPanelFactory.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/InvokeDialogListener.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/InvokeDialogListener.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/NonNegativeIntegerInputPanel.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/NonNegativeIntegerInputPanel.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/NullInputPanel.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/NullInputPanel.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ServiceConsoleCommandFactory.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/ServiceConsoleCommandFactory.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/StringInputPanel.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/StringInputPanel.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/access/AccessConsole.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/access/AccessConsole.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/management/ManagementConsole.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/console/management/ManagementConsole.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/datastream/DatastreamConduit.class
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream

Property changes on: fcrepo-client/fcrepo-client-admin/target/classes/org/fcrepo/client/datastream/DatastreamConduit.class
___________________________________________________________________
Added: svn:mime-type
   + application/octet-stream

Index: fcrepo-server/src/main/java/org/fcrepo/server/storage/ContentManagerParams.java
===================================================================
--- fcrepo-server/src/main/java/org/fcrepo/server/storage/ContentManagerParams.java	(revision 8829)
+++ fcrepo-server/src/main/java/org/fcrepo/server/storage/ContentManagerParams.java	(working copy)
@@ -1,5 +1,5 @@
 /* The contents of this file are subject to the license and copyright terms
- * detailed in the license directory at the root of the source tree (also 
+ * detailed in the license directory at the root of the source tree (also
  * available online at http://fedora-commons.org/license/).
  */
 package org.fcrepo.server.storage;
@@ -11,10 +11,10 @@
 
 
 /**
- * Simple data transfer object for the content manager. 
+ * Simple data transfer object for the content manager.
  * This should avoid breaking the content manager interface every
- * time the parameters change. 
- * 
+ * time the parameters change.
+ *
  * @version $Id$
  *
  */
@@ -26,11 +26,11 @@
     private String protocol;
     private boolean bypassBackend = false;
     private Context context;
-    
-    
+
+
     public ContentManagerParams(){
     }
-    
+
     public ContentManagerParams(String url, String mimeType, String username, String password){
         setUrl(url);
         this.mimeType = mimeType;
@@ -41,11 +41,11 @@
     public ContentManagerParams(String url){
         setUrl(url);
     }
-    
+
     public String getProtocol() {
         return protocol;
     }
-    
+
     public String getUrl() {
         return url;
     }
@@ -54,6 +54,9 @@
         try {
             this.protocol = new URL(url).getProtocol();
         } catch (MalformedURLException e) {
+            if(url.startsWith("irods://")) {
+        	return;
+            }
             throw new RuntimeException(e);
         }
     }
@@ -79,7 +82,7 @@
     public void setBypassBackend(boolean b) {
         bypassBackend = b;
     }
-    
+
     public boolean isBypassBackend() {
         return bypassBackend;
     }
Index: fcrepo-server/src/main/java/org/fcrepo/server/validation/ValidationUtility.java
===================================================================
--- fcrepo-server/src/main/java/org/fcrepo/server/validation/ValidationUtility.java	(revision 8829)
+++ fcrepo-server/src/main/java/org/fcrepo/server/validation/ValidationUtility.java	(working copy)
@@ -65,6 +65,8 @@
         } catch (MalformedURLException e) {
             if (url.startsWith(DatastreamManagedContent.UPLOADED_SCHEME)) {
                 return;
+            } else if (url.startsWith("irods://")) {
+                return;
             }
             throw new ValidationException("Malformed URL: " + url, e);
         }
Index: fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.eclipse.jdt.core.prefs
===================================================================
--- fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.eclipse.jdt.core.prefs	(revision 0)
+++ fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.eclipse.jdt.core.prefs	(revision 0)
@@ -0,0 +1,9 @@
+#Wed Oct 27 15:30:47 EDT 2010
+eclipse.preferences.version=1
+org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.6
+org.eclipse.jdt.core.compiler.compliance=1.6
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.problem.forbiddenReference=warning
+org.eclipse.jdt.core.compiler.source=1.6
Index: fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.eclipse.wst.common.component
===================================================================
--- fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.eclipse.wst.common.component	(revision 0)
+++ fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.eclipse.wst.common.component	(revision 0)
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project-modules id="moduleCoreId" project-version="1.5.0">
+    <wb-module deploy-name="fcrepo-webapp-fedora-3.4.2-SNAPSHOT">
+        <wb-resource deploy-path="/" source-path="/src/main/webapp"/>
+        <wb-resource deploy-path="/WEB-INF/classes" source-path="/src/main/resources"/>
+        <dependent-module deploy-path="/WEB-INF/lib" handle="module:/resource/fcrepo-server-3.4.2-SNAPSHOT/fcrepo-server-3.4.2-SNAPSHOT">
+            <dependency-type>uses</dependency-type>
+        </dependent-module>
+        <dependent-module deploy-path="/WEB-INF/lib" handle="module:/resource/fcrepo-common-3.4.2-SNAPSHOT/fcrepo-common-3.4.2-SNAPSHOT">
+            <dependency-type>uses</dependency-type>
+        </dependent-module>
+        <dependent-module deploy-path="/WEB-INF/lib" handle="module:/resource/fcrepo-generate-3.4.2-SNAPSHOT/fcrepo-generate-3.4.2-SNAPSHOT">
+            <dependency-type>uses</dependency-type>
+        </dependent-module>
+        <dependent-module deploy-path="/WEB-INF/lib" handle="module:/resource/fcrepo-security-jaas-3.4.2-SNAPSHOT/fcrepo-security-jaas-3.4.2-SNAPSHOT">
+            <dependency-type>uses</dependency-type>
+        </dependent-module>
+        <dependent-module deploy-path="/WEB-INF/lib" handle="module:/resource/fcrepo-security-pdp-3.4.2-SNAPSHOT/fcrepo-security-pdp-3.4.2-SNAPSHOT">
+            <dependency-type>uses</dependency-type>
+        </dependent-module>
+        <dependent-module deploy-path="/WEB-INF/lib" handle="module:/resource/fcrepo-security-pep-3.4.2-SNAPSHOT/fcrepo-security-pep-3.4.2-SNAPSHOT">
+            <dependency-type>uses</dependency-type>
+        </dependent-module>
+        <property name="context-root" value="fcrepo-webapp-fedora-3.4.2-SNAPSHOT"/>
+        <property name="java-output-path" value="/fcrepo-webapp-fedora-3.4.2-SNAPSHOT/target/classes"/>
+    </wb-module>
+</project-modules>
Index: fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.eclipse.wst.common.project.facet.core.xml
===================================================================
--- fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.eclipse.wst.common.project.facet.core.xml	(revision 0)
+++ fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.eclipse.wst.common.project.facet.core.xml	(revision 0)
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<faceted-project>
+  <installed facet="java" version="1.6"/>
+  <installed facet="jst.web" version="2.4"/>
+</faceted-project>
Index: fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.maven.ide.eclipse.prefs
===================================================================
--- fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.maven.ide.eclipse.prefs	(revision 0)
+++ fcrepo-webapp/fcrepo-webapp-fedora/.settings/org.maven.ide.eclipse.prefs	(revision 0)
@@ -0,0 +1,9 @@
+#Wed Oct 27 15:28:45 EDT 2010
+activeProfiles=
+eclipse.preferences.version=1
+fullBuildGoals=process-test-resources
+includeModules=false
+resolveWorkspaceProjects=true
+resourceFilterGoals=process-resources resources\:testResources
+skipCompilerPlugin=true
+version=1
Index: fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/base.css
===================================================================
--- fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/base.css	(revision 0)
+++ fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/base.css	(revision 0)
@@ -0,0 +1,15 @@
+body {
+  background-color: SteelBlue;
+  font-family: helvetica, arial, sans-serif;
+  font-size: 12pt;
+}
+
+.heading {
+  font-family: courier;
+  font-size: 16pt;
+  color: LightSteelBlue;
+  font-weight: bold;
+  text-align: center;
+  margin-top: 8px;
+  margin-bottom: 12px;
+}
Index: fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/error.xsl
===================================================================
--- fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/error.xsl	(revision 0)
+++ fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/error.xsl	(revision 0)
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<xsl:stylesheet version="2.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+
+<xsl:template match="/">
+<html>
+  <head><title>Query Error</title></head>
+  <body bgcolor="#ffeeee">
+    <table border="0" cellpadding="5">
+      <tr><td valign="top"><font size="+2"><u><b><xsl:value-of select="/error/message"/></b></u></font></td>
+      </tr>
+      <tr><td valign="top"><font size="-1"><pre><xsl:value-of select="/error/detail"/></pre></font></td>
+      </tr>
+    </table>
+  </body>
+</html>
+</xsl:template>
+
+</xsl:stylesheet>
\ No newline at end of file
Index: fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/form.css
===================================================================
--- fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/form.css	(revision 0)
+++ fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/form.css	(revision 0)
@@ -0,0 +1,58 @@
+.tabContent {
+  background-color: LightSteelBlue;
+  padding-top: 8px;
+  padding-left: 8px;
+  padding-right: 8px;
+}
+
+.inputForm {
+  padding-left: 6px;
+  padding-right: 6px;
+  padding-bottom: 6px;
+}
+
+.formTitle {
+  font-size: 14pt;
+  border-bottom: thin dashed #aaaaaa;
+  margin-bottom: 8px;
+}
+
+.tab-container {
+  background-color: LightSteelBlue;
+  border:1px solid #6D99B6;
+}
+
+.tabs { 
+  position:relative; 
+  height: 27px; 
+  margin: 0; 
+  padding: 0; 
+  background-color: SteelBlue;
+  overflow:hidden
+}
+
+.tabs li {
+  display:inline;
+}
+
+.tabs a:hover {
+  background-color: #ddddff;
+}
+
+.tabs a.tab-active {
+  background-color: LightSteelBlue;
+  border-right: 1px solid #6D99B6;
+}
+
+.tabs a  {
+  height: 27px; 
+  font:14px verdana, helvetica, sans-serif;font-weight:bold;
+  position:relative; 
+  padding:6px 10px 10px 10px; 
+  margin: 0px -4px 0px 0px; 
+  color:#000000;
+  text-decoration:none;
+  border-right:1px solid #6D99B6;
+}
+
+
Index: fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/form.xsl
===================================================================
--- fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/form.xsl	(revision 0)
+++ fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/form.xsl	(revision 0)
@@ -0,0 +1,321 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE trippi-form [
+<!ENTITY pageTitle            "Fedora Resource Index Query Service">
+<!ENTITY baseStylesheet       "base.css">
+<!ENTITY formStylesheet       "form.css">
+<!ENTITY tupleFormTitle       "Find Tuples">
+<!ENTITY tripleFormTitle      "Find Triples">
+<!ENTITY responseFormatPrompt "Response">
+<!ENTITY queryLanguagePrompt  "Language">
+<!ENTITY queryTextPrompt      "Query Text or URL">
+<!ENTITY usingTemplate        "(using template)">
+<!ENTITY templateTextPrompt   "Template Text or URL (if applicable)">
+<!ENTITY limitPrompt          "Limit">
+<!ENTITY otherPrompt          "Advanced">
+<!ENTITY distinctPrompt       "Force Distinct">
+<!ENTITY safeTypesPrompt      "Fake Media-Types">
+<!ENTITY streamPrompt         "Stream Immediately">
+]>
+<xsl:stylesheet version="2.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+
+<xsl:template match="/">
+<html>
+  <head>
+    <title>&pageTitle;</title>
+    <xsl:call-template name="linkStyle">
+      <xsl:with-param name="filename">&baseStylesheet;</xsl:with-param>
+    </xsl:call-template>
+    <xsl:call-template name="linkStyle">
+      <xsl:with-param name="filename">&formStylesheet;</xsl:with-param>
+    </xsl:call-template>
+<script language="JavaScript1.3"><![CDATA[
+var panes = new Array();
+
+function setupPanes(containerId, defaultTabId) {
+  // go through the DOM, find each tab-container
+  // set up the panes array with named panes
+  // find the max height, set tab-panes to that height
+  panes[containerId] = new Array();
+  var maxHeight = 0; var maxWidth = 0;
+  var container = document.getElementById(containerId);
+  var paneContainer = container.getElementsByTagName("div")[0];
+  var paneList = paneContainer.childNodes;
+  for (var i=0; i < paneList.length; i++ ) {
+    var pane = paneList[i];
+    if (pane.nodeType != 1) continue;
+    if (pane.offsetHeight > maxHeight) maxHeight = pane.offsetHeight;
+    if (pane.offsetWidth  > maxWidth ) maxWidth  = pane.offsetWidth;
+    panes[containerId][pane.id] = pane;
+    pane.style.display = "none";
+  }
+    paneContainer.style.height = maxHeight + "px";
+    paneContainer.style.width  = maxWidth + "px";
+    document.getElementById(defaultTabId).onclick();
+}
+
+function showPane(paneId, activeTab) {
+  // make tab active class
+  // hide other panes (siblings)
+  // make pane visible
+  
+    for (var con in panes) {
+    activeTab.blur();
+    activeTab.className = "tab-active";
+    if (panes[con][paneId] != null) { // tab and pane are members of this container
+      var pane = document.getElementById(paneId);
+      pane.style.display = "block";
+      var container = document.getElementById(con);
+      var tabs = container.getElementsByTagName("ul")[0];
+      var tabList = tabs.getElementsByTagName("a")
+      for (var i=0; i<tabList.length; i++ ) {
+        var tab = tabList[i];
+        if (tab != activeTab) tab.className = "tab-disabled";
+      }
+      for (var i in panes[con]) {
+        var pane = panes[con][i];
+        if (pane == undefined) continue;
+        if (pane.id == paneId) continue;
+        pane.style.display = "none"
+      }
+    }
+  }
+  return false;    
+}]]>
+</script>
+  </head>
+  <body onLoad="setupPanes('container1', 'tab1');">
+    <div class="heading">
+    &pageTitle;
+    </div>
+    <div class="tab-container" id="container1">
+      <ul class="tabs">
+        <li><a href="#" onClick="return showPane('tupleForm', this)" id="tab1">&tupleFormTitle;</a></li>
+        <li><a href="#" onClick="return showPane('tripleForm', this)">&tripleFormTitle;</a></li>
+        <li><a href="#" onClick="return showPane('aliases', this)">Show Aliases</a></li>
+      </ul>
+      <div class="tab-panes">  
+        <xsl:call-template name="doQueryForm"/>
+        <xsl:call-template name="doQueryForm">
+          <xsl:with-param name="triples">true</xsl:with-param>
+        </xsl:call-template>
+        <div id="aliases" class="tabContent">
+          <table border="0" cellpadding="3" cellspacing="3" width="100%">
+            <tr><td><u>Alias</u></td><td><u>URI Prefix</u></td></tr>
+            <xsl:for-each select="/query-service/alias-map/child::alias">
+              <tr>
+                <td><b><xsl:value-of select="@name"/></b></td>
+                <td><xsl:value-of select="@uri"/></td>
+              </tr>
+            </xsl:for-each>
+          </table>
+        </div>
+      </div>
+    </div>
+  </body>
+</html>
+</xsl:template>
+
+<xsl:template name="doQueryForm">
+  <xsl:param name="triples"/>
+  <div class="tabContent">
+    <xsl:attribute name="id">
+      <xsl:choose>
+        <xsl:when test="$triples">tripleForm</xsl:when>
+        <xsl:otherwise>tupleForm</xsl:otherwise>
+      </xsl:choose>
+    </xsl:attribute>
+    <form method="POST" target="TrippiQueryResults">
+      <xsl:attribute name="action">
+        <xsl:value-of select="/query-service/@href"/>
+      </xsl:attribute>
+      <input type="hidden" name="type">
+        <xsl:attribute name="value">
+          <xsl:choose>
+            <xsl:when test="$triples">triples</xsl:when>
+            <xsl:otherwise>tuples</xsl:otherwise>
+          </xsl:choose>
+        </xsl:attribute>
+      </input>
+      <table border="0" cellpadding="0" cellspacing="0">
+      <tr>
+        <td valign="top">
+        <xsl:choose>
+          <xsl:when test="$triples">
+              <xsl:call-template name="queryLanguageForm">
+              <xsl:with-param name="languages" select="/query-service/triple-languages"/>
+              <xsl:with-param name="otherLanguages" select="/query-service/tuple-languages"/>
+              <xsl:with-param name="appendText">&usingTemplate;</xsl:with-param>
+            </xsl:call-template>
+          </xsl:when>
+          <xsl:otherwise>
+            <xsl:call-template name="queryLanguageForm">
+              <xsl:with-param name="languages" select="/query-service/tuple-languages"/>
+            </xsl:call-template>
+          </xsl:otherwise>
+        </xsl:choose>
+      </td>
+      <td valign="top">
+      <xsl:choose>
+        <xsl:when test="$triples">
+          <xsl:call-template name="responseFormatForm">
+            <xsl:with-param name="formats" select="/query-service/triple-output-formats"/>
+          </xsl:call-template>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:call-template name="responseFormatForm">
+            <xsl:with-param name="formats" select="/query-service/tuple-output-formats"/>
+          </xsl:call-template>
+        </xsl:otherwise>
+      </xsl:choose>
+      </td>
+      <td valign="top"><xsl:call-template name="limitForm"/></td>
+      <td valign="top"><xsl:call-template name="otherForm"/></td>
+      </tr>
+      <tr><td colspan="4">
+      <xsl:choose>
+        <xsl:when test="$triples">
+          <table border="0" cellpadding="0" cellspacing="0">
+            <tr>
+              <td><xsl:call-template name="tripleQueryTextForm"/></td>
+              <td><xsl:call-template name="templateTextForm"/></td>
+            </tr>
+          </table>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:call-template name="tupleQueryTextForm"/>
+        </xsl:otherwise>
+      </xsl:choose>
+      </td></tr></table>
+      <center>
+      <div class="inputForm"><input type="submit" value=" Launch "/></div>
+      </center>
+    </form>
+  </div>
+
+</xsl:template>
+
+<xsl:template name="limitForm">
+  <div class="inputForm">
+    <fieldset>
+      <legend>&limitPrompt;</legend>
+    <select name="limit" size="4">
+      <option value="">Unlimited</option>
+      <option value="1000" selected="selected">1000</option>
+      <option value="100">100</option>
+      <option value="10">10</option>
+    </select>
+    </fieldset>
+  </div>
+</xsl:template>
+
+<xsl:template name="otherForm">
+  <div class="inputForm">
+    <fieldset>
+      <legend>&otherPrompt;</legend>
+      <input type="checkbox" name="distinct"/> <label for="distinct">&distinctPrompt;</label><br/>
+      <input type="checkbox" name="dt" checked="checked"/> <label for="dt">&safeTypesPrompt;</label><br/>
+      <input type="checkbox" name="stream"/> <label for="stream">&streamPrompt;</label>
+    </fieldset>
+  </div>
+</xsl:template>
+
+<xsl:template name="tupleQueryTextForm">
+  <div class="inputForm">
+    <fieldset>
+      <legend>&queryTextPrompt;</legend>
+    <textarea name="query" cols="80" rows="12" wrap="off"/>
+    </fieldset>
+  </div>
+</xsl:template>
+
+<xsl:template name="tripleQueryTextForm">
+  <div class="inputForm">
+    <fieldset>
+      <legend>&queryTextPrompt;</legend>
+    <textarea name="query" cols="36" rows="12" wrap="off"/>
+    </fieldset>
+  </div>
+</xsl:template>
+
+<xsl:template name="templateTextForm">
+  <div class="inputForm">
+    <fieldset>
+      <legend>&templateTextPrompt;</legend>
+    <textarea name="template" cols="36" rows="12" wrap="off"/>
+    </fieldset>
+  </div>
+</xsl:template>
+
+<xsl:template name="queryLanguageForm">
+  <xsl:param name="languages"/>
+  <xsl:param name="otherLanguages"/>
+  <xsl:param name="appendText"/>
+  <div class="inputForm">
+    <fieldset>
+      <legend>&queryLanguagePrompt;</legend>
+    <select name="lang" size="4">
+    <xsl:for-each select="$languages/child::language">
+      <option>
+        <xsl:attribute name="value">
+          <xsl:value-of select="@name"/>
+        </xsl:attribute>
+        <xsl:if test="position() = 1">
+          <xsl:attribute name="selected">selected</xsl:attribute>
+        </xsl:if>
+        <xsl:value-of select="@name"/>
+      </option>
+    </xsl:for-each>
+    <xsl:if test="$appendText">
+    <xsl:for-each select="$otherLanguages/child::language">
+      <option>
+        <xsl:attribute name="value">
+          <xsl:value-of select="@name"/>
+        </xsl:attribute>
+        <xsl:value-of select="@name"/>
+        <xsl:text> </xsl:text>
+        <xsl:value-of select="$appendText"/>
+      </option>
+    </xsl:for-each>
+    </xsl:if>
+    </select>
+    </fieldset>
+  </div>
+</xsl:template>
+
+<xsl:template name="responseFormatForm">
+  <xsl:param name="formats"/>
+  <div class="inputForm">
+    <fieldset>
+      <legend>&responseFormatPrompt;</legend>
+    <select name="format" size="4">
+    <xsl:for-each select="$formats/child::format">
+      <option>
+        <xsl:attribute name="value">
+          <xsl:value-of select="@name"/>
+        </xsl:attribute>
+        <xsl:if test="position() = 1">
+          <xsl:attribute name="selected">selected</xsl:attribute>
+        </xsl:if>
+        <xsl:value-of select="@name"/>
+      </option>
+    </xsl:for-each>
+    </select>
+    </fieldset>
+  </div>
+</xsl:template>
+
+<xsl:template name="linkStyle">
+  <xsl:param name="filename"/>
+  <link>
+    <xsl:attribute name="rel">stylesheet</xsl:attribute>
+    <xsl:attribute name="title">Default</xsl:attribute>
+    <xsl:attribute name="type">text/css</xsl:attribute>
+    <xsl:attribute name="href">
+      <xsl:value-of select="/query-service/@context"/>
+      <xsl:text>/</xsl:text>
+      <xsl:value-of select="$filename"/>
+    </xsl:attribute>
+  </link>
+</xsl:template>
+
+</xsl:stylesheet>
Index: fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/index.xsl
===================================================================
--- fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/index.xsl	(revision 0)
+++ fcrepo-webapp/fcrepo-webapp-fedora/target/classes/ri/index.xsl	(revision 0)
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<xsl:stylesheet version="2.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+
+<xsl:template match="/">
+<html>
+  <head><title>Trippi Server</title></head>
+  <body bgcolor="#eeeeff">
+  <center>
+    <table border="0" cellpadding="5">
+      <tr><td valign="top"><font size="+2"><u><b>Hosted Here:</b></u></font></td>
+      </tr>
+      <xsl:for-each select="/trippi-server/profile"><tr><td valign="top">
+      <table border="0" callpadding="5" cellspacing="10" bgcolor="#ddddff">
+      <tr>
+        <td valign="top">
+          <font size="+2">
+            <a>
+              <xsl:attribute name="href">
+                <xsl:value-of select="/trippi-server/@href"/>/<xsl:value-of select="@id"/>
+              </xsl:attribute>
+              <xsl:value-of select="@id"/>
+            </a>
+          </font><br/>
+          <font size="+1"><xsl:value-of select="@label"/></font><br/>
+          <i>Using <xsl:value-of select="@connector"/>.</i>
+        </td>
+        <td valign="top">
+        <u>Configuration</u>
+          <table border="0" cellpadding="0" cellspacing="0">
+            <xsl:for-each select="param">
+              <tr>
+                <td><font size="-1"><xsl:value-of select="@name"/></font></td>
+                <td>&#160;&#160;<font size="-1"><xsl:value-of select="@value"/></font></td>
+              </tr>
+            </xsl:for-each>
+          </table>
+        </td>
+      </tr>
+      </table>
+      </td></tr></xsl:for-each>
+    </table>
+    </center>
+  </body>
+</html>
+</xsl:template>
+
+</xsl:stylesheet>
\ No newline at end of file
